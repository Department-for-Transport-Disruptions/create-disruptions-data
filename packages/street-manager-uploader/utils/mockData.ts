const sqsMessage = {
    permitReferenceNumber: "TSR1591199404915-01",
    highwayAuthority: "CITY OF WESTMINSTER",
    highwayAuthoritySwaCode: 5990,
    worksLocationCoordinates: "LINESTRING(501251.53 222574.64,501305.92 222506.65)",
    streetName: "HIGH STREET NORTH",
    areaName: "LONDON",
    workCategory: "Standard",
    trafficManagementType: "Road closure",
    proposedStartDateTime: "2020-06-10T00:00:00.000Z",
    proposedEndDateTime: "2020-06-12T00:00:00.000Z",
    actualStartDateTime: null,
    actualEndDateTime: null,
    workStatus: "Works planned",
    usrn: "8401426",
    activityType: "Remedial works",
    worksLocationType: "Cycleway, Footpath",
    isTrafficSensitive: "Yes",
    permitStatus: "permit_modification_request",
    town: "LONDON",
    currentTrafficManagementType: "Multi-way signals",
    currentTrafficManagementTypeUpdateDate: null,
    lastUpdatedDateTime: "2020-06-04T08:00:00.000Z",
};

export const mockSqsEvent = {
    Records: [
        {
            messageId: "id",
            receiptHandle: "test",
            body: JSON.stringify(sqsMessage),
            md5OfBody: "test",
            eventSource: "test",
            eventSourceARN: "test",
            awsRegion: "test",
            attributes: {
                ApproximateReceiveCount: "test",
                SentTimestamp: "test",
                SenderId: "test",
                ApproximateFirstReceiveTimestamp: "test",
            },
            messageAttributes: {
                message: { dataType: "String" },
            },
        },
    ],
};
