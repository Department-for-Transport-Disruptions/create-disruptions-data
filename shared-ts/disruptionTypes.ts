import { z } from "zod";
import {
    consequenceOperatorsSchema,
    consequenceSchema,
    disruptionInfoSchema,
    disruptionSchema,
    networkConsequenceSchema,
    operatorConsequenceSchema,
    serviceSchema,
    servicesConsequenceSchema,
    stopSchema,
    stopsConsequenceSchema,
    validitySchema,
} from "./disruptionTypes.zod";

export type Disruption = z.infer<typeof disruptionSchema>;
export type DisruptionInfo = z.infer<typeof disruptionInfoSchema>;

export type Validity = z.infer<typeof validitySchema>;

export type Consequence = z.infer<typeof consequenceSchema>;
export type ConsequenceOperators = z.infer<typeof consequenceOperatorsSchema>;
export type Stop = z.infer<typeof stopSchema>;
export type Service = z.infer<typeof serviceSchema>;

export type NetworkConsequence = z.infer<typeof networkConsequenceSchema>;
export type OperatorConsequence = z.infer<typeof operatorConsequenceSchema>;
export type StopsConsequence = z.infer<typeof stopsConsequenceSchema>;
export type ServicesConsequence = z.infer<typeof servicesConsequenceSchema>;

export const routesSchema = z.object({
    inbound: z.array(stopSchema.partial()),
    outbound: z.array(stopSchema.partial()),
});

export type Routes = z.infer<typeof routesSchema>;
