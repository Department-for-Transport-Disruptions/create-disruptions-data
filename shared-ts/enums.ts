export enum SourceType {
    directReport = "directReport",
    email = "email",
    phone = "phone",
    fax = "fax",
    post = "post",
    feed = "feed",
    radio = "radio",
    tv = "tv",
    web = "web",
    pager = "pager",
    text = "text",
    other = "other",
}

export enum Progress {
    draft = "draft",
    open = "open",
    published = "published",
    closing = "closing",
    closed = "closed",
    pendingApproval = "pendingApproval",
    editPendingApproval = "editPendingApproval",
    draftPendingApproval = "draftPendingApproval",
    rejected = "rejected",
}

export enum MiscellaneousReason {
    accident = "accident",
    securityAlert = "securityAlert",
    congestion = "congestion",
    roadClosed = "roadClosed",
    incident = "incident",
    routeDiversion = "routeDiversion",
    unknown = "unknown",
    vandalism = "vandalism",
    overcrowded = "overcrowded",
    operatorCeasedTrading = "operatorCeasedTrading",
    vegetation = "vegetation",
    roadworks = "roadworks",
    specialEvent = "specialEvent",
    insufficientDemand = "insufficientDemand",
}

export enum PersonnelReason {
    unknown = "unknown",
    staffSickness = "staffSickness",
    staffInjury = "staffInjury",
    contractorStaffInjury = "contractorStaffInjury",
    staffAbsence = "staffAbsence",
    staffInWrongPlace = "staffInWrongPlace",
    staffShortage = "staffShortage",
    industrialAction = "industrialAction",
    unofficialIndustrialAction = "unofficialIndustrialAction",
    workToRule = "workToRule",
    undefinedPersonnelProblem = "undefinedPersonnelProblem",
}

export enum EquipmentReason {
    unknown = "unknown",
    pointsFailure = "pointsFailure",
    signalProblem = "signalProblem",
    trainWarningSystemProblem = "trainWarningSystemProblem",
    trackCircuitProblem = "trackCircuitProblem",
    signalFailure = "signalFailure",
    derailment = "derailment",
    engineFailure = "engineFailure",
    tractionFailure = "tractionFailure",
    breakDown = "breakDown",
    technicalProblem = "technicalProblem",
    brokenRail = "brokenRail",
    poorRailConditions = "poorRailConditions",
    wheelImpactLoad = "wheelImpactLoad",
    lackOfOperationalStock = "lackOfOperationalStock",
    defectiveFireAlarmEquipment = "defectiveFireAlarmEquipment",
    defectivePlatformEdgeDoors = "defectivePlatformEdgeDoors",
    defectiveCctv = "defectiveCctv",
    defectivePublicAnnouncementSystem = "defectivePublicAnnouncementSystem",
    ticketingSystemNotAvailable = "ticketingSystemNotAvailable",
    repairWork = "repairWork",
    constructionWork = "constructionWork",
    maintenanceWork = "maintenanceWork",
    emergencyEngineeringWork = "emergencyEngineeringWork",
    lateFinishToEngineeringWork = "lateFinishToEngineeringWork",
    powerProblem = "powerProblem",
    fuelProblem = "fuelProblem",
    swingBridgeFailure = "swingBridgeFailure",
    escalatorFailure = "escalatorFailure",
    liftFailure = "liftFailure",
    gangwayProblem = "gangwayProblem",
    closedForMaintenance = "closedForMaintenance",
    fuelShortage = "fuelShortage",
    deicingWork = "deicingWork",
    wheelProblem = "wheelProblem",
    luggageCarouselProblem = "luggageCarouselProblem",
    undefinedEquipmentProblem = "undefinedEquipmentProblem",
}

export enum EnvironmentReason {
    unknown = "unknown",
    fog = "fog",
    roughSea = "roughSea",
    heavySnowFall = "heavySnowFall",
    driftingSnow = "driftingSnow",
    blizzardConditions = "blizzardConditions",
    heavyRain = "heavyRain",
    strongWinds = "strongWinds",
    stormConditions = "stormConditions",
    stormDamage = "stormDamage",
    tidalRestrictions = "tidalRestrictions",
    highTide = "highTide",
    lowTide = "lowTide",
    ice = "ice",
    frozen = "frozen",
    hail = "hail",
    sleet = "sleet",
    highTemperatures = "highTemperatures",
    flooding = "flooding",
    waterlogged = "waterlogged",
    lowWaterLevel = "lowWaterLevel",
    highWaterLevel = "highWaterLevel",
    fallenLeaves = "fallenLeaves",
    fallenTree = "fallenTree",
    landslide = "landslide",
    undefinedEnvironmentalProblem = "undefinedEnvironmentalProblem",
    lightningStrike = "lightningStrike",
    sewerOverflow = "sewerOverflow",
    grassFire = "grassFire",
}

export enum DayType {
    monday = "monday",
    tuesday = "tuesday",
    wednesday = "wednesday",
    thursday = "thursday",
    friday = "friday",
    saturday = "saturday",
    sunday = "sunday",
}

export enum VehicleMode {
    bus = "bus",
    tram = "tram",
    ferryService = "ferryService",
    rail = "rail",
    underground = "underground",
}

export enum Severity {
    unknown = "unknown",
    normal = "normal",
    verySlight = "verySlight",
    slight = "slight",
    severe = "severe",
    verySevere = "verySevere",
}

export enum UserGroups {
    systemAdmins = "system-admins",
    orgAdmins = "org-admins",
    orgPublishers = "org-publishers",
    orgStaff = "org-staff",
    operators = "operators",
}

export enum PublishStatus {
    draft = "DRAFT",
    published = "PUBLISHED",
    editing = "EDITING",
    pendingApproval = "PENDING_APPROVAL",
    rejected = "REJECTED",
    editPendingApproval = "EDIT_PENDING_APPROVAL",
    pendingAndEditing = "PENDING_EDITING",
}

export enum SocialMediaPostStatus {
    pending = "Pending",
    successful = "Successful",
    rejected = "Rejected",
}

export enum Datasource {
    tnds = "tnds",
    bods = "bods",
}

export enum Modes {
    bus = "bus",
    coach = "coach",
    tram = "tram",
    ferry = "ferry",
    rail = "rail",
    underground = "underground",
    metro = "metro",
}
