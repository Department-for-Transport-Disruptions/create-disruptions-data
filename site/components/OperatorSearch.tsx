import kebabCase from "lodash/kebabCase";
import { ReactElement, useEffect, useState } from "react";
import Select, { ControlProps, GroupBase, OptionProps } from "react-select";
import FormElementWrapper, { FormGroupWrapper } from "./form/FormElementWrapper";
import { ErrorInfo } from "../interfaces";
import { Operator } from "../schemas/consequence.schema";

interface OperatorSearchProps<T> {
    operators: Operator[];
    stateUpdater: (change: string[], field: keyof T) => void;
    selectedOperatorNocs: string[];
    reset?: boolean;
    display: string;
    displaySize?: "s" | "m" | "l" | "xl";
    inputName: Extract<keyof T, string>;
    initialErrors?: ErrorInfo[];
}

export const sortOperatorByName = (operators: Operator[]): Operator[] => {
    return operators.sort((a, b) => (a.operatorPublicName > b.operatorPublicName ? 1 : -1));
};

const handleChange = <T,>(
    input: string[],
    inputName: keyof T,
    stateUpdater: (change: string[], field: keyof T) => void,
) => {
    stateUpdater(input, inputName);
};

const OperatorSearch = <T extends object>({
    display,
    displaySize = "m",
    operators,
    stateUpdater,
    selectedOperatorNocs,
    reset = false,
    initialErrors = [],
    inputName,
}: OperatorSearchProps<T>): ReactElement => {
    const [searchText, setSearchText] = useState("");

    useEffect(() => {
        if (reset) {
            handleChange<T>([], inputName, stateUpdater);
        }
    }, [reset, stateUpdater, inputName]);

    const controlStyles = (state: ControlProps<Operator, false, GroupBase<Operator>>) => ({
        fontFamily: "GDS Transport, arial, sans-serif",
        border: "#0b0c0c solid 2px",
        outline: state.isFocused ? "#ffdd00 solid 3px" : "none",
        color: state.isFocused ? "white" : "#0b0c0c",
        marginBottom: "20px",
        "&:hover": { borderColor: "#0b0c0c" },
        width: "75%",
    });

    const optionStyles = (state: OptionProps<Operator, false, GroupBase<Operator>>) => ({
        color: state.isFocused ? "white" : "#0b0c0c",
        backgroundColor: state.isFocused ? "#3399ff" : "white",
    });

    return (
        <FormGroupWrapper errorIds={[inputName]} errors={initialErrors}>
            <div className="govuk-form-group" id={kebabCase(inputName)}>
                <label className={`govuk-label govuk-label--${displaySize}`} htmlFor="operator-search-dropdown-value">
                    {display}
                </label>
                <FormElementWrapper errors={initialErrors} errorId={inputName} errorClass="govuk-input--error">
                    <Select
                        isSearchable
                        styles={{
                            control: (baseStyles, state) => ({
                                ...baseStyles,
                                ...controlStyles(state),
                            }),
                            option: (provided, state) => ({
                                ...provided,
                                ...optionStyles(state),
                            }),
                        }}
                        placeholder="Select operators"
                        getOptionLabel={(operator: Operator) => operator.operatorPublicName}
                        getOptionValue={(operator: Operator) => operator.nocCode}
                        options={sortOperatorByName(operators)}
                        onInputChange={(text) => {
                            setSearchText(text);
                        }}
                        inputValue={searchText}
                        onChange={(operator) => {
                            if (!!operator && !selectedOperatorNocs.find((noc) => noc === operator.nocCode)) {
                                handleChange([...selectedOperatorNocs, operator.nocCode], inputName, stateUpdater);
                            }
                        }}
                        id="operator-search"
                        instanceId="operator-dropdown"
                        inputId="operator-search-dropdown-value"
                        menuPlacement="auto"
                        menuPosition="fixed"
                        value={null}
                    />
                </FormElementWrapper>
            </div>
        </FormGroupWrapper>
    );
};

export default OperatorSearch;
