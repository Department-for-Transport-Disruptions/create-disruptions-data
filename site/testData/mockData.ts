/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { mockRequest, mockResponse } from "mock-req-res";
import { NextPageContext } from "next";
import React from "react";
import { Mock, vi } from "vitest";
import { COOKIE_ID_TOKEN, COOKIES_POLICY_COOKIE } from "../constants";

export interface GetMockContextInput {
    session?: { [key: string]: any };
    cookies?: any;
    body?: any;
    url?: any;
    uuid?: any;
    mockWriteHeadFn?: Mock<any, any>;
    mockEndFn?: Mock<any, any>;
    isLoggedin?: boolean;
    query?: any;
}

export const getMockRequestAndResponse = ({
    cookieValues = {},
    body = null,
    mockWriteHeadFn = vi.fn(),
    mockEndFn = vi.fn(),
    requestHeaders = {},
    isLoggedin = true,
    url = null,
    query = null,
}: GetMockRequestAndResponse = {}): { req: any; res: any } => {
    const res = mockResponse({ writeHead: mockWriteHeadFn, end: mockEndFn });

    const {
        idToken = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJjdXN0b206bm9jIjoiVEVTVCIsImVtYWlsIjoidGVzdEBleGFtcGxlLmNvbSIsImp0aSI6Ijg1MmQ1MTVlLTU5YWUtNDllZi1iMTA5LTI4YTRhNzk3YWFkNSIsImlhdCI6MTU5Mjk4NzMwNywiZXhwIjoxNTkyOTkwOTA3fQ.DFdxnpdhykDONOMeZMNeMUFpCHZ-hQ3UXczq_Qh0IAI",
        cookiePolicy = null,
    } = cookieValues;

    const defaultSession = {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        destroy: (): void => {},
    };

    let cookieString = "";

    cookieString += isLoggedin ? `${COOKIE_ID_TOKEN}=${idToken as string};` : "";

    cookieString += cookiePolicy ? `${COOKIES_POLICY_COOKIE}=${encodeURI(JSON.stringify(cookiePolicy))}` : "";

    const req = mockRequest({
        connection: {
            encrypted: true,
        },
        url,
        headers: {
            host: "localhost:3000",
            cookie: cookieString,
            origin: "localhost:3000",
            ...requestHeaders,
        },
        cookies: cookieValues,
        session: { ...defaultSession },
    });

    if (body) {
        req.body = body;
    }
    if (query) {
        req.query = query;
    }

    return { req, res };
};

export const getMockContext = ({
    cookies = {},
    body = null,
    uuid = {},
    mockWriteHeadFn = vi.fn(),
    mockEndFn = vi.fn(),
    isLoggedin = true,
    url = null,
    query = "",
}: GetMockContextInput = {}): NextPageContext => {
    const { req, res } = getMockRequestAndResponse({
        cookieValues: cookies,
        body,
        uuid,
        mockWriteHeadFn,
        mockEndFn,
        requestHeaders: {},
        isLoggedin,
        url,
    });

    const ctx: NextPageContext = {
        res,
        req,
        pathname: "",
        query,
        // eslint-disable-next-line react/display-name
        AppTree: () => React.createElement("div"),
    };

    return ctx;
};

export interface GetMockRequestAndResponse {
    cookieValues?: any;
    body?: any;
    uuid?: any;
    mockWriteHeadFn?: Mock<any, any>;
    mockEndFn?: Mock<any, any>;
    requestHeaders?: any;
    isLoggedin?: boolean;
    url?: any;
    query?: any;
}
