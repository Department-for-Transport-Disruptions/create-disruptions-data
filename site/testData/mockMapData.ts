export const mockMarkerData = [
    {
        atcoCode: "1800SB12581",
        commonName: "Cheapside",
        indicator: "Stop CL",
        longitude: -2.244551,
        latitude: 53.48096,
        bearing: "W",
        stopType: "BCT",
        busStopType: "MKD",
    },
    {
        atcoCode: "1800SB05801",
        commonName: "Cheapside",
        indicator: "Stop CO",
        longitude: -2.243361,
        latitude: 53.48097,
        bearing: "E",
        stopType: "BCT",
        busStopType: "MKD",
    },
    {
        atcoCode: "1800SB05751",
        commonName: "Manchester Town Hall",
        indicator: "Stop SC",
        longitude: -2.244697,
        latitude: 53.480024,
        bearing: "SE",
        stopType: "BCT",
        busStopType: "MKD",
    },
];
