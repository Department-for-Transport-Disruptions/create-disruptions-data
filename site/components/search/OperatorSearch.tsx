import { ConsequenceOperators } from "@create-disruptions-data/shared-ts/disruptionTypes";
import kebabCase from "lodash/kebabCase";
import { ReactElement, useEffect, useState } from "react";
import Select, { CSSObjectWithLabel, ControlProps, GroupBase, OptionProps } from "react-select";
import { ErrorInfo } from "../../interfaces";
import { Operator } from "../../schemas/consequence.schema";
import FormElementWrapper, { FormGroupWrapper } from "../form/FormElementWrapper";

interface OperatorSearchProps<T> {
    operators: Operator[];
    stateUpdater: (change: ConsequenceOperators[], field: keyof T) => void;
    selectedOperators: ConsequenceOperators[];
    reset?: boolean;
    display: string;
    displaySize?: "s" | "m" | "l" | "xl";
    inputName: Extract<keyof T, string>;
    initialErrors?: ErrorInfo[];
}

export const sortOperatorByName = (operators: Operator[]): Operator[] => {
    return operators.sort((a, b) => (a.operatorPublicName > b.operatorPublicName ? 1 : -1));
};

const handleChange = <T,>(
    input: ConsequenceOperators[],
    inputName: keyof T,
    stateUpdater: (change: ConsequenceOperators[], field: keyof T) => void,
) => {
    stateUpdater(input, inputName);
};

const OperatorSearch = <T extends object>({
    display,
    displaySize = "m",
    operators,
    stateUpdater,
    selectedOperators,
    reset = false,
    initialErrors = [],
    inputName,
}: OperatorSearchProps<T>): ReactElement => {
    const [searchText, setSearchText] = useState("");

    useEffect(() => {
        if (reset) {
            handleChange<T>([], inputName, stateUpdater);
        }
    }, [reset, stateUpdater, inputName]);

    const controlStyles = (state: ControlProps<Operator, false, GroupBase<Operator>>): CSSObjectWithLabel => ({
        fontFamily: "GDS Transport, arial, sans-serif",
        border: "#0b0c0c solid 2px",
        outline: state.isFocused ? "#ffdd00 solid 3px" : "none",
        color: state.isFocused ? "white" : "#0b0c0c",
        marginBottom: "20px",
        "&:hover": { borderColor: "#0b0c0c" },
        width: "75%",
    });

    const optionStyles = (state: OptionProps<Operator, false, GroupBase<Operator>>): CSSObjectWithLabel => ({
        color: state.isFocused ? "white" : "#0b0c0c",
        backgroundColor: state.isFocused ? "#3399ff" : "white",
    });

    return (
        <FormGroupWrapper errorIds={[inputName]} errors={initialErrors}>
            <div className="govuk-form-group" id={kebabCase(inputName)}>
                <label className={`govuk-label govuk-label--${displaySize}`} htmlFor="operator-search-dropdown-value">
                    {display}
                </label>
                <FormElementWrapper errors={initialErrors} errorId={inputName} errorClass="govuk-input--error">
                    <Select
                        isSearchable
                        styles={{
                            control: (baseStyles, state) => ({
                                ...baseStyles,
                                ...controlStyles(state),
                            }),
                            option: (provided, state) => ({
                                ...provided,
                                ...optionStyles(state),
                            }),
                        }}
                        placeholder="Select operators"
                        getOptionLabel={(operator: Operator) => `${operator.nocCode} - ${operator.operatorPublicName}`}
                        getOptionValue={(operator: Operator) => operator.nocCode}
                        options={sortOperatorByName(operators)}
                        onInputChange={(text) => {
                            setSearchText(text);
                        }}
                        inputValue={searchText}
                        onChange={(operator) => {
                            if (!!operator && !selectedOperators.find((noc) => noc.operatorNoc === operator.nocCode)) {
                                selectedOperators.push({
                                    operatorNoc: operator.nocCode,
                                    operatorPublicName: operator.operatorPublicName,
                                });
                                handleChange(selectedOperators, inputName, stateUpdater);
                            }
                        }}
                        id="operator-search"
                        instanceId="operator-dropdown"
                        inputId="operator-search-dropdown-value"
                        menuPlacement="auto"
                        menuPosition="fixed"
                        value={null}
                    />
                </FormElementWrapper>
            </div>
        </FormGroupWrapper>
    );
};

export default OperatorSearch;
