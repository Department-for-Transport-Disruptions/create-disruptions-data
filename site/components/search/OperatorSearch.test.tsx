import { OperatorConsequence } from "@create-disruptions-data/shared-ts/disruptionTypes";
import renderer from "react-test-renderer";
import { describe, it, expect, vi } from "vitest";
import OperatorSearch from "./OperatorSearch";
import { mockOperators } from "../../testData/mockData";

describe("OperatorSearch", () => {
    it("should render correctly with no errors", () => {
        const tree = renderer
            .create(
                <OperatorSearch<OperatorConsequence>
                    display="Operators impacted"
                    displaySize="l"
                    operators={mockOperators}
                    selectedOperators={[]}
                    stateUpdater={vi.fn()}
                    initialErrors={[]}
                    inputName="consequenceOperators"
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with errors", () => {
        const tree = renderer
            .create(
                <OperatorSearch<OperatorConsequence>
                    display="Operators impacted"
                    displaySize="l"
                    operators={mockOperators}
                    selectedOperators={[]}
                    stateUpdater={vi.fn()}
                    initialErrors={[{ errorMessage: "Select one or more operators", id: "consequence-operators" }]}
                    inputName="consequenceOperators"
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
