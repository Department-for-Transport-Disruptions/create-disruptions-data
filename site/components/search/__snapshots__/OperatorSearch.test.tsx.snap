// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OperatorSearch > should render correctly with errors 1`] = `
<DocumentFragment>
  <div
    class="govuk-form-group"
  >
    <div
      class="govuk-form-group"
      id="consequence-operators"
    >
      <label
        class="govuk-label govuk-label--l"
        for="operator-search-dropdown-value"
      >
        Operators impacted
      </label>
      <div
        class=""
      >
        <div
          class="text-lg text-black css-b62m3t-container"
          id="operator-search"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-operator-dropdown-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
            role="log"
          />
          <div
            class=" css-1efjb9t-control"
          >
            <div
              class=" css-1fdsijx-ValueContainer"
            >
              <div
                class=" css-yl2wa5-placeholder"
                id="react-select-operator-dropdown-placeholder"
              >
                Select operators
              </div>
              <div
                class=" css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-describedby="react-select-operator-dropdown-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="operator-search-dropdown-value"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`OperatorSearch > should render correctly with no errors 1`] = `
<DocumentFragment>
  <div
    class="govuk-form-group"
  >
    <div
      class="govuk-form-group"
      id="consequence-operators"
    >
      <label
        class="govuk-label govuk-label--l"
        for="operator-search-dropdown-value"
      >
        Operators impacted
      </label>
      <div
        class=""
      >
        <div
          class="text-lg text-black css-b62m3t-container"
          id="operator-search"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-operator-dropdown-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
            role="log"
          />
          <div
            class=" css-1efjb9t-control"
          >
            <div
              class=" css-1fdsijx-ValueContainer"
            >
              <div
                class=" css-yl2wa5-placeholder"
                id="react-select-operator-dropdown-placeholder"
              >
                Select operators
              </div>
              <div
                class=" css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-describedby="react-select-operator-dropdown-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="operator-search-dropdown-value"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
